### ISSUE TITLE
restricting tekton.dev labels in metadata

### ISSUE THREAD

**Issue by pritidesai** [tektoncd/pipeline#4366]
Tekton resources are automatically assigned certain labels including the ones with `tekton.dev`. The labels are created and propagated from one resource to another using this group name to signify the association and relationship between pipeline task, taskRun, pipeline, and pipelineRun . We should restrict possibility of overwriting those labels otherwise it could result in conflicting resources. 

Result of the discussion in PR #2826.

Opening this issue to track the work done in PR #2891 and possibly implementing the restriction.

**Comment by austinzhao-go** [comment]
thanks @afrittoli hints out. 
[TEP-0106](https://github.com/tektoncd/community/pull/695) should also be applied with the similar constraints to avoid overwriting the Tekton metadata. marked a follow up action on my side to sync the work from this thread.

**Comment by lbernick** [comment]
@vdemeester is this closed now as a result of https://github.com/tektoncd/pipeline/pull/6441?

**Comment by vdemeester** [comment]
@lbernick not really, #6441 was just about the "other projects" labels issues (chains.tekton.dev, …). This issue is more about restricting `tekton.dev/*` labels / annotations to only be set by us (tektoncd/pipeline controllers) and filtered from outside (aka not allowed).

**Comment by vdemeester** [comment]
Unfreezing this issue — it's been frozen since 2023 but the gap is still real. There is **no webhook validation** preventing users from overwriting `tekton.dev/*` labels and annotations on Tekton resources. PR #2891 was closed without merge, and PR #6441 only addressed `chains.tekton.dev` labels.

This could cause subtle bugs if a user sets conflicting `tekton.dev/pipeline` or `tekton.dev/task` labels. Marking as `priority/important-longterm` — needs a champion to implement admission validation.

/remove-lifecycle frozen
